lightning/src/ln/channelmanager.rs

```diff
@@ -1267,22 +1267,30 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 
 	/// Creates a new outbound channel to the given remote node and with the given value.
 	///
-	/// user_id will be provided back as user_channel_id in FundingGenerationReady events to allow
-	/// tracking of which events correspond with which create_channel call. Note that the
-	/// user_channel_id defaults to 0 for inbound channels, so you may wish to avoid using 0 for
-	/// user_id here. user_id has no meaning inside of LDK, it is simply copied to events and
-	/// otherwise ignored.
+	/// `user_id` will be provided back as `user_channel_id` in [`Event::FundingGenerationReady`]
+	/// to allow tracking of which events correspond with which `create_channel` call. Note that
+	/// the `user_channel_id` defaults to 0 for inbound channels, so you may wish to avoid using 0
```

Comment:
Seems like `user_channel_id` is suited to be an `Option`, which can be `None` for inbound channels

Or an enum
```
UserChannelId {
    Outbound(u64),
    Inbound
}
```

Reply:
Hmm, can we make that a todo in #1125?

Reply:
Fine by me